models/cohortdata.go

```diff
@@ -130,20 +130,45 @@ func (h CohortData) RetrieveCohortOverlapStats(sourceId int, caseCohortId int, c
 	otherFilterConceptIds []int64, filterCohortPairs []utils.CustomDichotomousVariableDef) (CohortOverlapStats, error) {
 
+	// special case for when filter lists are empty:
+	if len(otherFilterConceptIds) == 0 && len(filterCohortPairs) == 0 {
+		// call the faster version of overlap check:
+		return h.RetrieveCohortOverlapStatsWithoutFiltering(sourceId, caseCohortId, controlCohortId)
+	}
+
 	var dataSourceModel = new(Source)
 	omopDataSource := dataSourceModel.GetDataSource(sourceId, Omop)
 	resultsDataSource := dataSourceModel.GetDataSource(sourceId, Results)
 
 	// count persons that are in the intersection of both case and control cohorts, filtering on filterConceptValue:
 	var cohortOverlapStats CohortOverlapStats
-	query := omopDataSource.Db.Table(omopDataSource.Schema+".observation_continuous as observation").
+	query := omopDataSource.Db.Table(omopDataSource.Schema+".observation_continuous as observation"+omopDataSource.GetViewDirective()).
 		Select("count(distinct(observation.person_id)) as case_control_overlap").
 		Joins("INNER JOIN "+resultsDataSource.Schema+".cohort as case_cohort ON case_cohort.subject_id = observation.person_id").
 		Joins("INNER JOIN "+resultsDataSource.Schema+".cohort as control_cohort ON control_cohort.subject_id = case_cohort.subject_id"). // this one allows for the intersection between case and control and the assessment of the overlap
 		Where("case_cohort.cohort_definition_id = ?", caseCohortId).
 		Where("control_cohort.cohort_definition_id = ?", controlCohortId)
 
-	query = QueryFilterByConceptIdsAndCohortPairsHelper(query, otherFilterConceptIds, filterCohortPairs, omopDataSource.Schema, resultsDataSource.Schema)
+	query = QueryFilterByConceptIdsAndCohortPairsHelper(query, otherFilterConceptIds, filterCohortPairs, omopDataSource, resultsDataSource.Schema)
+
+	meta_result := query.Scan(&cohortOverlapStats)
+	return cohortOverlapStats, meta_result.Error
+}
+
+// Basically the same as the method above, but without any filtering on any concepts or on any CustomDichotomousVariableDef:
+func (h CohortData) RetrieveCohortOverlapStatsWithoutFiltering(sourceId int, caseCohortId int, controlCohortId int) (CohortOverlapStats, error) {
+
+	var dataSourceModel = new(Source)
+	omopDataSource := dataSourceModel.GetDataSource(sourceId, Omop)
+	resultsDataSource := dataSourceModel.GetDataSource(sourceId, Results)
+
+	// count persons that are in the intersection of both case and control cohorts, filtering on filterConceptValue:
+	var cohortOverlapStats CohortOverlapStats
+	query := omopDataSource.Db.Table(resultsDataSource.Schema+".cohort as case_cohort").
+		Select("count(distinct(case_cohort.subject_id)) as case_control_overlap").
+		Joins("INNER JOIN "+resultsDataSource.Schema+".cohort as control_cohort ON control_cohort.subject_id = case_cohort.subject_id"). // this one allows for the intersection between case and control and the assessment of the overlap
```

Comment:
We will try this, the only other thing to think about is whether a simple `INTERSECT` statement would be more efficient or if they are the same. Just making the note here

Reply:
the code now contains INTERSECT statements for the HARE breakdown endpoint.